sync/trypush.py

```diff
@@ -450,6 +451,71 @@ def delete(self):
                 idx.delete(key, data)
 
 
+class TcPush(TryPush):
```

Comment:
So, now having actually looked at the code, I'm not convinced that this is the right approach.

We typically only store these `ProcessData`-derived objects in cases where we are actually managing the lifecycle of some process. In this case we aren't but just intereacting with it at a point. That means a lof of the API here is unused; in practice we only call `download_logs`.

Unless we are planning to use a lot more of this API surface in the future, it seems like we should just factor out the `download_logs` method into a function and, if we want to store the taskgroup_id, put it on the sync object directly.